src/adagrams.js

```diff
@@ -1,15 +1,219 @@
+// HELPER FUNCTIONS
+
+/*
+This function uses the built in method 'reduce' to reduce the array based on the parameters supplied.
+It looks to see if the letter is in the array and, if it is, it adds 1 to the return value.
+*/
+export const countTimes = (arr, val) => arr.reduce((a, v) => (v === val ? a + 1 : a), 0);
+
+// This function loads the winner into an object
+export const loadWinner = (word, score) => {
+  let winner = {};
+
+  winner.word = word;
+  winner.score = score;
+  
+  return winner;
+};
+
+// This function handles any tie scores
+export const tieBreaker = maxScore => {
+  let smallestWordLen = 1000;
+  let winner = {};
+  let nxtWord, winWord = '';
+  let nxtScore, winScore = 0;
+  
+  for (let i = 0; i < maxScore.length; i++) {
+      nxtWord = maxScore[i][0];
+      nxtScore = maxScore[i][1];
+
+      if (nxtWord.length === 10) {
+          winWord = nxtWord;
+          winScore = nxtScore;
+          break;
+      } else if (nxtWord.length < smallestWordLen) {
+          smallestWordLen = nxtWord.length;
+          winWord = nxtWord;
+          winScore = nxtScore;
+      } else {
+          continue;    
+      };
+  };
+
+  winner = loadWinner(winWord, winScore);
+  return winner;
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const letterPool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  let clone = JSON.parse(JSON.stringify(letterPool));
+  let userHand = [];
+
+  while (userHand.length != 10) {
+    // Using the random math generator as the index for a random letter from the pool
+    const randNum = Math.floor(Math.random() * 26);
```

Comment:
instead of hardcoding 26 you could also multiple by the letterPool length just in case it were to change